goserver/server/server_test.go

```diff
@@ -76,3 +75,28 @@ func TestLaunchHandlerCheckISSAuthorization(t *testing.T) {
 		})
 	}
 }
+
+func TestHandleLaunch(t *testing.T) {
+	fhirURL := "https://authorized.fhir.com"
+	sm := session.NewManager(sessiontest.NewMemoryStore(), func() string { return "test-session-id" }, 30*time.Minute)
+	s, err := NewServer(fhirURL, 0, sm)
+	if err != nil {
+		t.Fatalf("NewServer(authorizedFHIRURL, 0, sm) -> %v, nil expected", err)
+		return
+	}
+	rr := httptest.NewRecorder()
+	req := httptest.NewRequest("GET", "/?iss="+fhirURL, nil)
+	s.handleLaunch(rr, req)
```

Comment:
Is it possible to avoid calling the method directly here and instead invoke this logic by making an actual (fake) HTTP request? This will make the test more realistic as we'll be including any integration issues between the HTTP library and our code.

Reply:
The usage examples in the httptest package: https://godoc.org/net/http/httptest#example-Server use HTTP client libraries, so it seems recommended. All HTTP traffic just happens on localhost so I don't think there would be any issues here (many of our internal unit tests do this exact thing when testing RPC servers).

Reply:
I see. It's using httptest.NewServer with a passed in handler. (I tried to use http.Server and my Mac asked me for permission everytime I ran the test)